zerver/tests/test_upload.py

```diff
@@ -1673,16 +1679,21 @@ def test_delete_avatar_image(self) -> None:
         avatar_medium_path_id = avatar_path_id + "-medium.png"
 
         self.assertEqual(user.avatar_source, UserProfile.AVATAR_FROM_USER)
-        self.assertIsNotNone(bucket.get_key(avatar_path_id))
-        self.assertIsNotNone(bucket.get_key(avatar_original_image_path_id))
-        self.assertIsNotNone(bucket.get_key(avatar_medium_path_id))
+        self.assertIsNotNone(bucket.Object(avatar_path_id))
+        self.assertIsNotNone(bucket.Object(avatar_original_image_path_id))
+        self.assertIsNotNone(bucket.Object(avatar_medium_path_id))
 
         zerver.lib.actions.do_delete_avatar_image(user)
 
         self.assertEqual(user.avatar_source, UserProfile.AVATAR_FROM_GRAVATAR)
-        self.assertIsNone(bucket.get_key(avatar_path_id))
-        self.assertIsNone(bucket.get_key(avatar_original_image_path_id))
-        self.assertIsNone(bucket.get_key(avatar_medium_path_id))
+        # Boto3 won't return a None type from trying to get a non-existent key,
+        # it just raises a ClientError
+        with self.assertRaises(botocore.exceptions.ClientError):
+            bucket.Object(avatar_path_id).load()
+        with self.assertRaises(botocore.exceptions.ClientError):
+            bucket.Object(avatar_original_image_path_id).load()
+        with self.assertRaises(botocore.exceptions.ClientError):
+            bucket.Object(avatar_medium_path_id).load()
```

Comment:
This probably isn't the right way, but I also lost coverage having this in a single `with` statement.  At the moment, I can't think of a better way to check that the keys are no longer there.

Reply:
A single `with` statement wouldn’t be right anyway. That would only test that one of the three lines raises an exception.